xbmc/cores/RetroPlayer/rendering/VideoRenderers/RPWinRenderer.cpp

```diff
@@ -0,0 +1,216 @@
+/*
+ *      Copyright (C) 2017 Team Kodi
+ *      http://kodi.tv
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this Program; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "RPWinRenderer.h"
+#include "cores/RetroPlayer/rendering/RPRenderFactory.h"
+#include "cores/VideoPlayer/VideoRenderers/VideoShaders/WinVideoFilter.h" //! @todo
+#include "guilib/GraphicContext.h"
+#include "guilib/D3DResource.h"
+#include "utils/log.h"
+#include "windowing/WindowingFactory.h"
+
+extern "C" {
+#include "libswscale/swscale.h"
+}
+
+#include <cstring>
+
+using namespace KODI;
+using namespace RETRO;
+
+CRPBaseRenderer* CRPWinRenderer::Create()
+{
+  return new CRPWinRenderer();
+}
+
+void CRPWinRenderer::Register()
+{
+  CRPRendererFactory::RegisterRenderer(CRPWinRenderer::Create);
+}
+
+CRPWinRenderer::CRPWinRenderer() :
+  m_bQueued(false),
+  m_intermediateTarget(new CD3DTexture),
+  m_outputShader(new COutputShader)
+{
+  // Initialize CRPBaseRenderer
+  m_scalingMethod = VS_SCALINGMETHOD_LINEAR;
+}
+
+CRPWinRenderer::~CRPWinRenderer()
+{
+  Deinitialize();
+  if (m_swsContext)
+    sws_freeContext(m_swsContext);
+}
+
+bool CRPWinRenderer::Configure(AVPixelFormat format, unsigned int width, unsigned int height, unsigned int orientation)
+{
+  m_format = format;
+  m_sourceWidth = width;
+  m_sourceHeight = height;
+  m_renderOrientation = orientation;
+
+  m_targetPixFormat = AV_PIX_FMT_BGRA;
+  m_targetDxFormat = DXGI_FORMAT_B8G8R8X8_UNORM;
+  if (g_Windowing.IsFormatSupport(DXGI_FORMAT_B8G8R8A8_UNORM, D3D11_USAGE_DYNAMIC))
+    m_targetDxFormat = DXGI_FORMAT_B8G8R8A8_UNORM;
```

Comment:
three lines above may be replaced with
`m_targetDxFormat = g_Windowing.GetBackBuffer()->GetFormat();`

Reply:
Done